### PR TITLE
[feature] Added connect under reset to stlink_open_usb( )

### PR DESCRIPTION
**Pull request by geoffreymbrown**
I've added a small change to stlink_open_usb to support connect under reset.  In addition, I added an option to tests/usb.c  (--reset or --no-reset) to use this mode.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "cmake.configureOnOpen": false
+}
```
Comment (Nightwalker-87) [inline .vscode/settings.json:3]: Note: We don't need this IDE-specific config in the codebase. Can you remove it?

**Comment by geoffreymbrown** [comment]
Mea culpa.  There are times when vscode does too much.
Geoffrey

File: .vscode/settings.json (hunk 1 above)
Comment (geoffreymbrown) [reply .vscode/settings.json]: Very sorry !   I've removed it.

Comment (Nightwalker-87) [reply .vscode/settings.json]: Thx.

**Comment by chenguokai** [comment]
As far as I have seen, the code added a new argument to `tests/usb.c`, not to the existing utilities like `st-flash`, while the original issue refers to `st-flash`. Would you be kind enough to add this argument for `st-flash`?

Relevant files are `tools/flash_opts.c`, `tools/flash.c` and some relevant headers.

**Comment by geoffreymbrown** [comment]
I can but I need a bit more guidance.   What flag should I add ?
--connect-under-reset ?

 Since the flash tool halts the processor in any case, the right behavior
is probably
to

1) connect under reset
2) force debug
3) release the reset
4) reread the chip info (under reset, the flash size isn't returned
properly)

Should I make those changes ?   I have limited ways to test them, but they
would be localized to that option flag.

Geoffrey

**Comment by chenguokai** [comment]
@Nightwalker-87 Maybe you are the best person to make decisions on this kind of issues:-)

**Comment by geoffreymbrown** [comment]
I'll do it in the next couple of days.   Probably should do the same for
st-info.

**Comment by Nightwalker-87** [comment]
> I can but I need a bit more guidance. What flag should I add ? --connect-under-reset ?

I think that is what @chenguokai referred to. You can proceed as proposed, but we will need someone to test it before merging.

**Comment by geoffreymbrown** [comment]
I've added the additional code to my branch -- the other pull request was
still in progress, so I'm not quite sure how to proceed.   I tested the
code with an stm32l476RG nucleo board executing the attached small program
which blinks a bit and then goes to standby mode with the debugger
interface disabled.  Without the --connect-under-reset flag both st-info
and st-flash fail.   With the flag they both execute successfully.

The code is (relatively) non-invasive.  Other than checking for the flag
and passing it to the usb open routine, it doesn't do anything.  I did have
to extend the usb open code a bit to force debug mode and release the reset
line in order to minimize the changes to st-info and st-flash

**Comment by Nightwalker-87** [comment]
@geoffreymbrown: This PR has been updated automatically with the update to your branch. Thx for testing and the related description. Looks good to me. :+1:

**Comment by Nightwalker-87** [comment]
@chenguokai: Can you review the updated changes as well?

**Comment by Nightwalker-87** [comment]
Thx for your contribution @geoffreymbrown.

**Pull request merged**